examples/proto.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/python3
+
+from ubuntu_image.image import Diagnostics, GiB, Image
+
+# Create empty image of a fixed size
+# TODO: compute rough good size after seeing all the required snaps.
+
+image = Image('img', GiB(4))
+
+# Install GRUB to MBR
+# TODO: this has to be represented in the image.yaml
+# NOTE: the boot.img has to be a part of the gadget snap itself
+# FIXME: embed a pointer to 2nd stage in bios-boot partition
+image.copy_blob('blogs/img.mbr', bs=446, count=1, conv='notrunc')
+
+# Create BIOS boot partition
+#
+# The partition is 1MiB in size, as recommended by various partitioning guides.
+# The actual required size is much, much smaller.
+#
+# https://www.gnu.org/software/grub/manual/html_node/BIOS-installation.html#BIOS-installation
+image.partition(new='1:4MiB:+1MiB')
```

Comment:
I think we want to separate the index, the offset and the size in the constructor. The particular `sgdisk` directive is easily computed from the internal representation. Please disregard this comment if this already happens below (reading top-to-bottom).